lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java

```diff
@@ -1,118 +1,159 @@
+import pro.taskana.workbasket.api.models.WorkbasketSummary;
+import pro.taskana.workbasket.internal.builder.WorkbasketAccessItemBuilder;
+
+@TaskanaIntegrationTest
+class CompleteTaskAccTest {
+
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject WorkbasketService workbasketService;
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject CurrentUserContext currentUserContext;
+
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasketSummary;
+  ObjectReference defaultObjectReference;
+
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void setup() throws Exception {
+    defaultClassificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService);
+    defaultWorkbasketSummary = defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasketSummary.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+
+    defaultObjectReference = defaultTestObjectReference().build();
+  }
 
   @WithAccessId(user = "user-1-1")
   @Test
-  void testCompleteTask() throws Exception {
-    assertThat(taskService.getTask("TKI:000000000000000000000000000000000001").getState())
-        .isEqualTo(TaskState.CLAIMED);
-    Task completedTask = taskService.completeTask("TKI:000000000000000000000000000000000001");
+  void should_CompleteClaimedTask() throws Exception {
+    Task claimedTask = createTaskClaimedByUser().buildAndStore(taskService);
+
+    Task completedTask = taskService.completeTask(claimedTask.getId());
+
     assertThat(completedTask).isNotNull();
     assertThat(completedTask.getCompleted()).isNotNull();
     assertThat(completedTask.getState()).isEqualTo(TaskState.COMPLETED);
-    assertThat(completedTask.getModified()).isNotEqualTo(completedTask.getCreated());
+    assertThat(completedTask.getCompleted())
+        .isEqualTo(completedTask.getModified())
+        .isNotNull()
+        .isNotEqualTo(completedTask.getCreated());
   }
 
   @WithAccessId(user = "admin")
   @Test
-  void should_completeClaimedTaskByAnotherUser_When_UserIsAdmin() throws Exception {
-    assertThat(taskService.getTask("TKI:000000000000000000000000000000000029").getState())
-        .isEqualTo(TaskState.CLAIMED);
-    Task completedTask = taskService.completeTask("TKI:000000000000000000000000000000000029");
+  void should_CompleteClaimedTaskByAnotherUser_When_UserIsAdmin() throws Exception {
+    Task claimedTask = createTaskClaimedByUser().buildAndStore(taskService);
+
+    Task completedTask = taskService.completeTask(claimedTask.getId());
+
     assertThat(completedTask).isNotNull();
     assertThat(completedTask.getCompleted()).isNotNull();
```

Comment:
since the assert statements two lines below already contains `isNotNull` this is not necessary.